testsupport/tiers/tier_setup.go

```diff
@@ -0,0 +1,189 @@
+		require.NoError(t, err)
+	}
+
+	// custom values
+	for _, modify := range modifiers {
+		err := modify(hostAwait, tier)
+		require.NoError(t, err)
+	}
+	err := hostAwait.CreateWithCleanup(context.TODO(), tier.NSTemplateTier)
+	require.NoError(t, err)
+	return tier
+}
+
+// createCustomNSTemplateTier updates the given "tier" using the modifiers
+// returns the latest version of the NSTemplateTier
+func UpdateCustomNSTemplateTier(t *testing.T, hostAwait *HostAwaitility, tier *CustomNSTemplateTier, modifiers ...CustomNSTemplateTierModifier) *CustomNSTemplateTier {
+	// reload the underlying NSTemplateTier resource before modifying it
+	tmplTier, err := hostAwait.WaitForNSTemplateTier(tier.Name)
+	require.NoError(t, err)
+	tier.NSTemplateTier = tmplTier
+	// make sure we have the very latest version of the given tier (to avoid the update conflict on the server-side)
+	for _, modify := range modifiers {
+		err := modify(hostAwait, tier)
+		require.NoError(t, err)
+	}
+	err = hostAwait.Client.Update(context.TODO(), tier.NSTemplateTier)
+	require.NoError(t, err)
+	return tier
+}
+
+func duplicateTierTemplate(hostAwait *HostAwaitility, namespace, tierName, origTemplateRef string) (string, error) {
+	origTierTemplate := &toolchainv1alpha1.TierTemplate{}
+	if err := hostAwait.Client.Get(context.TODO(), test.NamespacedName(hostAwait.Namespace, origTemplateRef), origTierTemplate); err != nil {
+		return "", err
+	}
+	newTierTemplate := &toolchainv1alpha1.TierTemplate{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      fmt.Sprintf("%sfrom%s", tierName, origTierTemplate.Name),
+			Labels:    map[string]string{"producer": "toolchain-e2e"},
+		},
+		Spec: origTierTemplate.Spec,
+	}
+	newTierTemplate.Spec.TierName = tierName
+	if err := hostAwait.CreateWithCleanup(context.TODO(), newTierTemplate); err != nil {
+		if !errors.IsAlreadyExists(err) {
+			return "", err
+		}
+	}
+	return newTierTemplate.Name, nil
+}
+
+func MoveUserToTier(t *testing.T, hostAwait *HostAwaitility, username, tierName string) {
+	t.Logf("moving user '%s' to tier '%s'", username, tierName)
+	changeTierRequest := NewChangeTierRequest(hostAwait.Namespace, username, tierName)
+	err := hostAwait.CreateWithCleanup(context.TODO(), changeTierRequest)
+	require.NoError(t, err)
+	_, err = hostAwait.WaitForChangeTierRequest(changeTierRequest.Name, toBeComplete)
+	require.NoError(t, err)
+}
+
+func NewChangeTierRequest(namespace, murName, tier string) *toolchainv1alpha1.ChangeTierRequest {
+	return &toolchainv1alpha1.ChangeTierRequest{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:    namespace,
+			GenerateName: "changetierrequest-",
+		},
+		Spec: toolchainv1alpha1.ChangeTierRequestSpec{
+			MurName:  murName,
+			TierName: tier,
+		},
+	}
+}
+
+func DeactivationTimeoutDays(timeoutDurationDays int) TierModifier {
+	return func(tier *toolchainv1alpha1.NSTemplateTier) error {
+		tier.Spec.DeactivationTimeoutDays = timeoutDurationDays
+		return nil
+	}
+}
```

Comment:
These functions looks like a duplication of what is in `testsupport/tier_setup.go` file